server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -3105,19 +3109,20 @@ private Pair<List<ServiceOfferingJoinVO>, Integer> searchForServiceOfferingsInte
         Integer cpuSpeed = cmd.getCpuSpeed();
         Boolean encryptRoot = cmd.getEncryptRoot();
 
+        final Account owner = _accountMgr.finalizeOwner(caller, accountName, domainId, projectId);
         SearchCriteria<ServiceOfferingJoinVO> sc = _srvOfferingJoinDao.createSearchCriteria();
-        if (!_accountMgr.isRootAdmin(caller.getId()) && isSystem) {
-            throw new InvalidParameterValueException("Only ROOT admins can access system's offering");
+        if (!_accountMgr.isRootAdmin(owner.getId()) && isSystem) {
```

Comment:
@RodrigoDLopez 
I did not look into it yet,
I think in some lines, we should use `caller` , not `owner`

Reply:
I'll check further and get back to you.